### DISCUSSION TITLE
the source code and the solution-modules

### DISCUSSION THREAD

**Discussion by dinani65** [jishnujayakumar/MLRC2020-EmbedKGQA#10] [Q&A]
The main code includes 3 folders: RoBERTa, LSTM and train_embeddings and the paper-solution contains four main steps: KG embedding, Question embedding, Answer selection and Relation matching.
As far as I have understood, Relation matching has not been implemented, LSTM is related to question embedding for MetaQA dataset, RoBERTa is related to WebQSP dataset. what about the train_embeddings folder?
Could u please explain more regarding the relations between codes and the modules of the solution?

**Comment by jishnujayakumar** [comment]
It is used for training MetaQA KG dataset whereas LibKGE is used for training WebQSP KG dataset. Two different methods are used for the two KG datasets. For further information, visit [here](https://www.github.com/jishnujayakumar/MLRC2020-EmbedKGQA/tree/main/train_embeddings%2FREADME.md).

**Reply by sridharkonduri-99** [reply]
I have a further query on this, did you try RoBERTa implementation on MetaQA dataset? My implementation had huge training time with really bad accuracy

**Reply by jishnujayakumar** [reply]
No, we didn't try RoBERTa on the MetaQA dataset.

**Comment by dinani65** [comment]
I do apologize it due to my question is not related to the topic but I would greatly happy if u can answer.
what is the best and fast way to generate embeddings of some freebase-IDs (e.g. RDF2Vec technique)? Do I have to download the whole freebase dataset at first and use a graph database?
